pkg/util/kubernetes/kubelet/kubelet.go

```diff
@@ -168,6 +178,13 @@ func (ku *KubeUtil) GetNodeInfo(ctx context.Context) (string, string, error) {
 
 // GetNodename returns the nodename of the first pod.spec.nodeName in the PodList
 func (ku *KubeUtil) GetNodename(ctx context.Context) (string, error) {
+	if ku.useAPIServer {
+		stats, err := ku.GetLocalStatsSummary(ctx)
```

Comment:
Question: where else is `GetNodename()` being called? Would it makes sense to add a check on `ku.kubeletClient.config.nodeName` and just return that value if it's set?